components/BattleScreen/BattleScreen.js

```diff
@@ -36,15 +40,34 @@ export default function BattleScreen() {
 			}
 
 			const damageTaken = Math.floor(Math.random() * (50 - 1 + 1) + 1);
-			setPlayerHealth(playerHealth - damageTaken);
+			setPlayerHealth(Math.max(playerHealth - damageTaken, 0)); // Schaden wird maximal auf die aktuelle Gesundheit des Spielers angewendet
 			setIsDisabled(false); // Button wird wieder aktiviert
+			setEnemyAttacking(true);
 
 			if (playerHealth - damageTaken <= 0) {
 				setVictory(false);
 			}
+			console.log("Damage taken:", damageTaken);
 		}, 1000);
+		console.log("Damage dealt:", actualDamage);
 	};
 
+	useEffect(() => {
+		if (playerAttacking) {
+			setTimeout(() => {
+				setPlayerAttacking(false);
+			}, 500);
+		}
+	}, [playerAttacking]);
+
+	useEffect(() => {
+		if (enemyAttacking) {
+			setTimeout(() => {
+				setEnemyAttacking(false);
+			}, 500);
+		}
+	}, [enemyAttacking]);
+
```

Comment:
Why don't you put these into the same `useEffect`?

Reply:
yes true, is more beautiful, modified.